# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/systemservices/objc_video_capture/video_capture_avfoundation.h
# Commit: c07caf074089
# Full Hash: c07caf07408914b5514f89c3d46fb1dced9fef06
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2023-01-13 21:39:47
# Regressor Bug: 1806605
# File Overlap Count: 1
# Description:
#   Bug 1806605 - Pass TrackingId instead of nsCString to CaptureStage. r=padenot
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D165824
# ==============================================================================

diff -r 731dcbc83284 -r c07caf074089 dom/media/systemservices/objc_video_capture/video_capture_avfoundation.h
--- a/dom/media/systemservices/objc_video_capture/video_capture_avfoundation.h	Fri Jan 13 14:25:00 2023 +0000
+++ b/dom/media/systemservices/objc_video_capture/video_capture_avfoundation.h	Fri Jan 13 14:25:00 2023 +0000
@@ -42,7 +42,7 @@
   // Callback. This can be called on any thread.
   int32_t OnFrame(webrtc::VideoFrame& aFrame) MOZ_EXCLUDES(api_lock_);
 
-  void SetTrackingId(const char* _Nonnull aTrackingId) MOZ_EXCLUDES(api_lock_) override;
+  void SetTrackingId(uint32_t aTrackingIdProcId) MOZ_EXCLUDES(api_lock_) override;
 
   // Allows the capturer to start the recording before calling OnFrame, to cover more operations
   // under the same measurement.
@@ -60,7 +60,7 @@
   // thread only.
   mozilla::Maybe<VideoCaptureCapability> mCapability MOZ_GUARDED_BY(api_lock_);
   // Id string uniquely identifying this capture source. Written on the mChecker thread only.
-  mozilla::Maybe<nsCString> mTrackingId MOZ_GUARDED_BY(api_lock_);
+  mozilla::Maybe<mozilla::TrackingId> mTrackingId MOZ_GUARDED_BY(api_lock_);
   // Adds frame specific markers to the profiler while mTrackingId is set.
   mozilla::PerformanceRecorderMulti<mozilla::CaptureStage> mPerformanceRecorder;
   std::atomic<ProfilerThreadId> mCallbackThreadId;